### ISSUE TITLE
Argonaut Json parser causes ClassCastException

### ISSUE THREAD

**Issue by julien-truffaut** [http4s/http4s#157]
I wrote 2 `EntityDecoder`, one using `ArgonautSupport.json` and one using `argonaut.Parse`. The latter is working but the first one generates the following error: java.lang.ClassCastException: scala.Some cannot be cast to argonaut.Json

```scala
def jsonEntityDecoder[A](implicit ev: DecodeJson[A]): EntityDecoder[A] =
    ArgonautSupport.json.map(json =>
      ev.decodeJson(json).fold(
        (message, history) => sys.error(s"Could not decode json $json, error: $message, cursor: $history"),
        identity
      )
    )

  def jsonEntityDecoder2[A](implicit ev: DecodeJson[A]): EntityDecoder[A] =
    EntityDecoder.decodeBy(MediaType.`application/json`){msg =>
      org.http4s.DecodeResult.apply(EntityDecoder.decodeString(msg).map{s =>
        Parse.parse(s).fold(
          m => ParseResult.fail(s"Cannot parse string $s into json, error: $m"),
          ParseResult.success
        ).flatMap(json =>
          ev.decodeJson(json).fold(
            (m, h) => ParseResult.fail(s"Cannot parse json $json, error: $m $h"),
            ParseResult.success
          )
        )
      })
    }
```

I am sure we can reproduce the error without DecodeJson but I didn't have the time to do so

**Comment by rossabaker** [comment]
Fix released as v0.5.2, on its way to Maven Central.

**Comment by julien-truffaut** [comment]
thanks @rossabaker

**Comment by rossabaker** [comment]
Released v0.5.3 with the above commit.
